src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -353,7 +353,7 @@ void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const Qg
   for ( auto it = layers.constBegin(); it != layers.constEnd(); ++it )
   {
     QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( it.value() );
-    if ( !layer || !layer->flags().testFlag( QgsMapLayer::Searchable ) )
```

Comment:
shouldn't we use isValid instead?

Reply:
If isValid() does a data provider check, then we could. The problem here is that the creation of a new vector layer feature source blindly assumes that the passed on vector layer has a data provider: https://github.com/qgis/QGIS/blob/96dd42b1501d1d139ca6ef24fa2a6f43df79047b/src/core/qgsvectorlayerfeatureiterator.cpp#L34

Reply:
ok